### ISSUE TITLE
`_public_bitwise_less_than` for `ActiveProtocolSuite`?

### ISSUE THREAD

**Issue by matthiasgeihs** [cicada-mpc/cicada-mpc#82]
I've noticed that `_public_bitwise_less_than` is available for `AdditiveProtocolSuite` and `ShamirProtocolSuite`, but not for `ActiveProtocolSuite`.

However, it seems like it is quite straightforward to port the code by simply using `ActiveArrayShare` for all secret-shared values (instead of `AdditiveArrayShare` or `ShamirArrayShare`). I actually went ahead and did this and the resulting protocol seems to give the correct results.

**Question:** Am I right to assume that this simple port of `_public_bitwise_less_than` to `ActiveProtocolSuite` as described above will just work and be secure?

**Comment by tshead2** [comment]
I'll leave it to Ken to address your correctness question, but in the meantime I assume that you're wanting to use _public_bitwise_less_than for your own work?  We'd need to make those methods public (in the software engineering sense) in that case.  Apologies if you already know this, but by convention Python functions with names that begin with an underscore are considered private implementation details that are subject to change or removal at any time.

Cheers,
Tim

**Comment by matthiasgeihs** [comment]
Yes, I am aware of that convention. At the moment it's just prototyping, but, yes, it would be useful to have these part of the public interface for my use case. Actually, the core function that I need is not the comparison, but deterministic truncation of high and low bits.

**Comment by kgoss1729** [comment]
I apologize for my delayed responses. I have multiple projects at present and am very busy so I am not able to be as responsive as I would normally like. 

The issue, iirc, due to the way in which we've handled these functionalities in the active protocol suite which generally just call their respective implementations in additive and shamir. Since that was the case we never had a cause to implement _public_bitwise_less_than in the Active suite as, when needed, it just calls the private implementations in either additive or shamir. I do not see at present any risk of doing as you propose given the same caveats as are spelled out in the docstrings of the private implementations in additive and shamir.

We do very much appreciate your interest and engagement, please bear with us, and me in particular, as I try to find time to meaningfully respond to the issues you're raising in my tight time constraints.

**Comment by matthiasgeihs** [comment]
Absolutely no rush needed. Thanks for the info!

Feel free to close the issue, if you want to take that up another time.
